src/common/container/lock_free_array.cpp

```diff
@@ -107,6 +108,20 @@ bool LOCK_FREE_ARRAY_TYPE::Contains(const ValueType &value) const {
   return exists;
 }
 
+template <typename ValueType>
```

Comment:
Two main concerns: 
1. What happens if another thread modifies the value between line 115 and 117?
2. This seems inefficient. What is the use case? Would using a `CuckooHash` be more efficient?